[WIP] v6 Docs Dev (jschalz) (#2025)
loft-sh/devspace: docs/pages/configuration/images/build-engines/kaniko.mdx

    kaniko:
      cache: false
```
**Explanation:**

> - The first image `backend` would be built using kaniko and make use of the build cache.

Reviewer: ```suggestion
- The first image `backend` is built using kaniko and makes use of the build cache.
```

> - The second image `frontend` would be built using kaniko and **not** use the build cache.

Reviewer: ```suggestion
- The second image `frontend` is built using kaniko and does **not** use the build cache.
```

    kaniko:
      snapshotMode: time
```
**Explanation:**

> - The first image `backend` would be built using kaniko and creating full filesystem snapshots.

Reviewer: ```suggestion
- The first image `backend` is built using kaniko and creates full filesystem snapshots.
```

> - The second image `frontend` would be built using kaniko and calculate filesystem snapthots only based on `mtime`.

Reviewer: ```suggestion
- The second image `frontend` is built using kaniko and calculates filesystem snapshots only based on `mtime`.
```

    kaniko:
      insecure: false
```
**Explanation:**

> The image `backend` would be built using kaniko and pushing to the insecure registry `123.456.789.0:5000` would be allowed.

Reviewer: ```suggestion
The image `backend` is built using kaniko and pushes to the insecure registry `123.456.789.0:5000` if allowed.
```

      - --cache-dir=/tmp
      - --verbosity=debug
```
**Explanation:**

> The image `backend` would be built using kaniko and the flags `--cache-dir=/tmp --verbosity=debug` would be set when running the build command within the kaniko pod used for image building.

Reviewer: ```suggestion
The image `backend` is built using kaniko and the flags `--cache-dir=/tmp --verbosity=debug` are set when running the build command within the kaniko pod used for image building.
```

    kaniko:
      pullSecret: custom-pull-secret
```
**Explanation:**

> The image `backend` would be built using kaniko and kaniko would use the Kubernete secret `custom-pull-secret` to pull images from registries that require authentication.

Reviewer: ```suggestion
The image `backend` is built using kaniko. Kaniko uses the Kubernete secret, `custom-pull-secret`, to pull images from registries that require authentication.
```

:::tip Creating Pull Secrets using `pullSecrets`

> If you use the `pullSecret` option, you have to create the secret before DevSpace tries to build your images. You can automate this by [using the pullSecrets feature](../../pullSecrets/basics.mdx) as shown in this example:

Reviewer: ```suggestion
If you use the `pullSecret` option, you must create the secret before DevSpace tries to build your images. You can automate this by [using the pullSecrets feature](../../pullSecrets/basics.mdx) as shown in this example:
```

          - key: aws-token
            path: token.json
```
**Explanation:**

> The above configuration would create a kaniko container which mounts the following volumes:

Reviewer: ```suggestion
The above configuration creates a kaniko container which mounts the following volumes:
```

    kaniko:
      namespace: build-namespace
```
**Explanation:**

> The image `backend` would be built using kaniko and the build pod started to run the kaniko build process would be created within the namespace `build-namespace` within the cluster that the current kube-context points to.

Reviewer: ```suggestion
The image `backend` is built using kaniko. The build pod starts to run the kaniko build process and creates the namespace `build-namespace` within the cluster that the current kube-context points to.
```
